hathor/manager.py

```diff
@@ -1076,7 +1096,8 @@ def tx_fully_validated(self, tx: BaseTransaction) -> None:
         # Publish to pubsub manager the new tx accepted, now that it's full validated
         self.pubsub.publish(HathorEvents.NETWORK_NEW_TX_ACCEPTED, tx=tx)
 
-        self.tx_storage.indexes.mempool_tips.update(tx)
+        if self.tx_storage.indexes.mempool_tips:
```

Comment:
Isn't it worth implementing a mock for each index type? They would do nothing, just ignoring the calls. This would remove a lot of conditionals in our code. What do you think?

Reply:
Not for _each_ index type. Some indexes have method signatures where the return value is really important, like returning `True/False` to indicate if it's a new addition or now. Maybe for _each optional_ index, although that would hide some possible problems if for some reason the index is not added when it should have been, and instead of failing an assert the mock call would silently continue which would cause problems (particularly for sync-v2, the deps indexes must work correctly in order for the sync to advance).

Reply:
You're right. It's not a good idea in this case. Thanks!